### ISSUE TITLE
evaluation

### ISSUE THREAD

**Issue by sky186** [JDAI-CV/fast-reid#166]
@L1aoXingyu 
您好作者， 评估模型  报错
./tools/train_net.py --config-file ./configs/Market1501/bagtricks_R50.yml --eval-only \
MODEL.WEIGHTS .log/model_0012119.pth MODEL.DEVICE "cuda:0"

attributeerro: attempt to set weight "........pth" but  CfgNode is immutable

**Comment by sky186** [comment]
@L1aoXingyu 
评估代码，我想简单一点测试，不想一堆类的调用，所以抽取一个通用接口，我也不加Checkpointer(model).load(cfg.MODEL.WEIGHTS)   hooks.PreciseBN  这些操作

模型读取加载预训练参数，
    from fastreid.modeling.backbones import build_resnet_backbone
    cfg = setup(yaml_path)
    cfg.defrost()
    cfg.MODEL.BACKBONE.PRETRAIN = False
    model = build_resnet_backbone(cfg)  
    # model = nn.DataParallel(model)
    model = model.cuda()
    model.load_state_dict((torch.load(model_path)["model"]), strict=False)
提取特征 模型输入的是 读取的图片 data， 然后 cos距离计算，
这样特征计算送给 map函数 cmc 函数，结果是不对的，

我的理解哪里有问题吗，模型读取预训练参数这么写好像有问题，我不知道哪里有问题。

**Comment by L1aoXingyu** [comment]
我这里并没有出现这个问题。

**Comment by sky186** [comment]
@L1aoXingyu 
这个问题我也不知道怎么引起，后来同样的环境命令，运行又正常了
我自己改的代码 加载数据和代码里面的就是 不一样0-1之间的tensor
但是我打印您的代码，这里的是 0-255 的图片输入，我看了这边的totensor 不一致,
我想问下您这里数据输入的详细处理是什么？

resize 和 ToTensor
但是代码里面的 

我想问下代码里面的数据处理问题
    test_transform = transforms.Compose([
        transforms.Resize((256, 128), interpolation=3),
        transforms.ToTensor(), #ToTensor：convert a PIL image to tensor (H*W*C) in range [0,255] to a torch.Tensor(C*H*W) in the range [0.0,1.0]
        # transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])# 加载模型前向计算数据处理，有这个操作，不是在这里实现的
    ])

![图片](https://user-images.githubusercontent.com/56809648/87275040-df508980-c50f-11ea-9e27-2d6525e594c1.png)

**Comment by L1aoXingyu** [comment]
`ToTensor` 用的下面这个函数，就是把图片通道的顺序换一下

https://github.com/JDAI-CV/fast-reid/blob/eb880767140f1be490dc796f650374d1115f9bb3/fastreid/data/transforms/functional.py#L12
